api/opds.py

```diff
@@ -1431,13 +1431,15 @@ def active_loans_for(
         return response
 
     @classmethod
-    def single_item_feed(cls, circulation, item, fulfillment=None, test_mode=False,
+    def single_item_feed(cls, circulation, item, license_pool=None, fulfillment=None, test_mode=False,
                          feed_class=AcquisitionFeed, **response_kwargs):
         """Construct a response containing a single OPDS entry representing an active loan
         or hold.
 
         :param circulation: A CirculationAPI
         :param item: A Loan or Hold -- perhaps one that was just created or looked up.
+        :param license_pool: An optional LicensePool object required only
+            when item is None (there is no a Loan or Hold).
```

Comment:
If you make it possible for `item` to be a LicensePool (it's already polymorphic between a Loan and a Hold) you won't have to change the method signature.

Reply:
Sorry, I didn't follow your advice for the first time, you're absolutely right, it's better to have a single argument covering all the three cases.